app/scripts/portland_ingest.py

```diff
@@ -0,0 +1,158 @@
+    riders. The riders value was originally provided as an average by season 
+    and day type (Saturday, Sunday, and Weekday). In order to fit the data model
+    for the project which is at the datetime level, the data has been mapped to dates in 2023.
+"""
+
+import json
+import sys
+import os
+from requests.models import Response
+import django
+import datetime
+from datetime import datetime
+import pytz
+from typing import List, Dict, Any
+from django.db.utils import IntegrityError
+
+current_dir = os.path.dirname(os.path.abspath(__file__))
+parent_dir = os.path.abspath(os.path.join(current_dir, ".."))
+sys.path.append(parent_dir)
+
+os.environ.setdefault("DJANGO_SETTINGS_MODULE", "geodjango.settings")
+
+django.setup()
+
+
+from route_rangers_api.models import TransitStation, RidershipStation
+
+
+def get_transit_station_ids_pdx() -> list:
+    """
+    Get a list of all id values from the TransitStation table for PDX
+    """
+    return list(TransitStation.objects.filter(city="PDX").values_list("id", flat=True))
+
+
+def format_input_ridership_data(
+    json_file_path: str, start_date: datetime.date, end_date: datetime.date
+) -> List[Dict[str, Any]]:
+    """
+    Format the dates in the input JSON ridership data
+    and filter the data set to only include the dates that we need to
+    work with
+    """
+    formatted_data = []
+    with open(json_file_path, "r") as file:
+        data = json.load(file)
+    for record in data:
+        timestamp_seconds = int(record["date"]) / 1000
+        date_obj = datetime.fromtimestamp(timestamp_seconds)
+        formatted_date = date_obj.strftime("%Y-%m-%d %H:%M:%S")
+        record["date"] = formatted_date
+        if start_date <= date_obj <= end_date:
+            formatted_data.append(record)
+    print("JSON ready..")
+    return formatted_data
+
+
+def ingest_pdx_ridership_data(
+    json_file_path: str,
+    start_date: datetime.date,
+    end_date: datetime.date,
+    list_of_db_ids: List,
+) -> None:
+    """
+    Ingest portland ridership data from extracted JSON file
+    into Django database for a given subset of date ranges
+    """
+    stations_not_matched = []
+
+    formatted_data = format_input_ridership_data(json_file_path, start_date, end_date)
+
+    for record in formatted_data:
+        try:
+            # Get the id from TransitStation using station_id + city = 'PDX'
+            transit_station = TransitStation.objects.filter(
+                station_id=str(record["station_id"]), city="PDX"
+            ).first()
+
+            if transit_station:
+                foreign_key = transit_station.id
+                date_str = record["date"]
+                date_obj = datetime.strptime(date_str, "%Y-%m-%d %H:%M:%S")
+                # date formatting to match model
+
+                if start_date <= date_obj <= end_date:
+                    # Check if foreign key exists in the list of TransitStation ids
+                    if foreign_key in list_of_db_ids:
```

Comment:
Given the use that you are given to `list_of_db_ids` I don't think is necessary to create the object as you can check if an object with the id exists by just filtering (as you do on line 94).

Also, since you are retrieving a `TranistStation `object  in line 94, if it exists I don't think it's necessary to check if the foreign key exists in the `list_of_db_ids` as it always should (we are basically retrieving an id from an object from the table and then checking if the id is in the table again). I think this could be adding some inefficiency as it adds one operation on every object inserted.